apps/api/e2e/tests/auth.test.ts

```diff
@@ -0,0 +1,254 @@
+  const domain = new URL(baseUrl).hostname || "localhost";
+
+  // Admin client for creating teams
+  let adminClient: ApiClient;
+  let adminApiKey: string;
+
+  beforeEach(async () => {
+    await cleanupTestState();
+    // Create admin account directly using the setup endpoint
+    const response = await axios.post(`${baseUrl}/api/admin/setup`, {
+      username: ADMIN_USERNAME,
+      password: ADMIN_PASSWORD,
+      email: ADMIN_EMAIL,
+    });
+
+    // Store the admin API key for authentication
+    adminApiKey = response.data.admin.apiKey;
+    expect(adminApiKey).toBeDefined();
+
+    // Create admin client
+    adminClient = createTestClient();
+    const loginSuccess = await adminClient.loginAsAdmin(adminApiKey);
+    expect(loginSuccess).toBe(true);
+  });
+
+  it("should complete the full SIWE authentication flow", async () => {
+    // Register a team
+    const teamName = `Team ${Date.now()}`;
+    const email = `team${Date.now()}@example.com`;
+    const contactPerson = "Test Contact";
+
+    const { team, apiKey } = await registerTeamAndGetClient(
+      adminApiKey,
+      teamName,
+      email,
+      contactPerson,
+    );
+    expect(team).toBeDefined();
+    expect(team.id).toBeDefined();
+    expect(team.name).toBe(teamName);
+    expect(team.email).toBe(email);
+    expect(team.contactPerson).toBe(contactPerson);
+    expect(apiKey).toBeDefined();
+
+    // Create a session client (without API key)
+    const sessionClient = new ApiClient(undefined, baseUrl);
+
+    // 1. Get a nonce
+    const nonceResponse = (await sessionClient.getNonce()) as
+      | NonceResponse
+      | ErrorResponse;
+    expect(nonceResponse).not.toHaveProperty("error");
+    const { nonce } = nonceResponse as NonceResponse;
+    expect(nonce).toBeDefined();
+    expect(typeof nonce).toBe("string");
+
+    // 2. Create a SIWE message
+    const message = await createSiweMessage(domain, nonce);
+
+    // 3. Sign the message
+    const signature = await signMessage(message);
+
+    // 4. Login with the signed message
+    const loginResponse = (await sessionClient.login(message, signature)) as
+      | LoginResponse
+      | ErrorResponse;
+    expect(loginResponse).not.toHaveProperty("error");
+    const loginData = loginResponse as LoginResponse;
+    // Use case-insensitive match for wallet address (Ethereum addresses can have mixed case)
+    expect(loginData.wallet.toLowerCase()).toBe(
+      testWalletAddress.toLowerCase(),
+    );
+
+    // 5. Logout
+    const logoutResponse = (await sessionClient.logout()) as
+      | LogoutResponse
+      | ErrorResponse;
+    expect(logoutResponse).not.toHaveProperty("error");
+    expect((logoutResponse as LogoutResponse).message).toBe(
+      "Logged out successfully",
+    );
+  });
+
+  it("should allow access to protected endpoints after SIWE login", async () => {
+    // Register a team with our test wallet address
+    const teamResult = await adminClient.registerTeam(
```

Comment:
@dtbuchholz I might be missing some understanding here, but it seems like this test is a almost the same as the one above.
Since `testWallet` is always used (here and in the helpers), the only difference is this test is using the SDK and the other is using Axios.  Is that the reason for two tests?

Reply:
Ok, no worries if you want to leave it, but maybe rename it?